sdk/core/azure-core/tests/perf_tests/upload_binary.py

```diff
@@ -0,0 +1,77 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from time import time
+from wsgiref.handlers import format_date_time
+from devtools_testutils.perfstress_tests import get_random_bytes, WriteStream
+
+from azure.core.exceptions import (
+    HttpResponseError,
+    map_error,
+)
+from azure.storage.blob._generated.operations._block_blob_operations import (
+    build_upload_request
+)
+from ._test_base import _BlobTest
+
+
+class UploadBinaryDataTest(_BlobTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        blob_name = "uploadtest"
+        self.blob_endpoint = f"{self.account_endpoint}{self.container_name}/{blob_name}"
+        self.error_map = {
+            401: ClientAuthenticationError,
+            404: ResourceNotFoundError,
+            409: ResourceExistsError,
+            304: ResourceNotModifiedError,
+        }
+
+    async def global_setup(self):
+        await super().global_setup()
+        self.data = get_random_bytes(self.args.size)
+
+    def run_sync(self):
+        current_time = format_date_time(time())
+        request = build_upload_request(
+            url=self.blob_endpoint,
+            content=self.data,
```

Comment:
We should also test a streaming upload - i.e. passing in a readable.

Reply:
Storage is reading from the stream under the hood, so copied that over here. Is this what you were thinking for streaming upload?

Reply:
So I looked through the repo and can't find any example of services that pass the readable through directly, rather than chunking. Do you happen to know of any?

Reply:
OOB: should work just passing stream directly to content. Was not working b/c of issue here, but is resolved: https://github.com/Azure/azure-sdk-for-python/issues/33997